presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java

```diff
@@ -764,17 +764,21 @@ public int read(byte[] buffer, int offset, int length)
                 int bytesRead = retry()
                         .maxAttempts(maxAttempts)
                         .exponentialBackoff(new Duration(1, TimeUnit.SECONDS), maxBackoffTime, maxRetryTime, 2.0)
-                        .stopOn(InterruptedException.class, UnrecoverableS3OperationException.class)
+                        .stopOn(InterruptedException.class, UnrecoverableS3OperationException.class, AbortedException.class)
                         .onRetry(STATS::newReadRetry)
                         .run("readStream", () -> {
                             seekStream();
                             try {
                                 return in.read(buffer, offset, length);
                             }
-                            catch (Exception e) {
-                                STATS.newReadError(e);
+                            catch (AbortedException abortedException) {
```

Comment:
I don't think we need this catch block at all. If we don't catch the exception, the `stopOn()` addition will prevent retry. This could also be thrown by `seekStream()`.

We also don't want to mislead the call by returning `0` (which indicates EOF) when the read actually failed.

Closing the stream isn't necessary, since the user of the `InputStream` should always call the `close()` method, even on failure.